### PR TITLE
Feature/docker image

### PR DESCRIPTION
**Pull request by sokovninn**
This PR includes:

- Adding a Dockerfile.
- Adding GHCR publish action.
- Adding GAR publish action.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,13 @@ To install with pip:
 pip install datadreamer
 ```
 
+Docker build and run:
+
+```bash
+docker build -t datadreamer .
+docker run -v "$(pwd):/app" datadreamer --save_dir generated_dataset
```
Comment (kozlov721) [inline README.md:82]: On my laptop I also have to add `--runtime nvidia`, this might be good to mention as a sidenote.

Comment (tersekmatija) [reply README.md:82]: Can we also add it without build (to use from GHCR directly)?

**Review by HonzaCuhel — COMMENTED** [review]
LGTM, the package is not yet published to ghcr, right @sokovninn?

I wasn't able to run the dockerized version on a Mac with M chipset even when building the container with flag `--platform linux/arm64`. Since the arm architecture can be troublesome with cuda,  so I'd maybe add a note to the Docker installation that it is recommended for Linux users (and Windows probably as well)?

**Comment by sokovninn** [comment]
> LGTM, the package is not yet published to ghcr, right @sokovninn?

@HonzaCuhel Yes, it will be published after merging to main one and release.

> I wasn't able to run the dockerized version on a Mac with M chipset even when building the container with flag `--platform linux/arm64`. Since the arm architecture can be troublesome with cuda, so I'd maybe add a note to the Docker installation that it is recommended for Linux users (and Windows probably as well)?

Good point. So you were not able to run even on CPU?

**Comment by HonzaCuhel** [comment]
> Good point. So you were not able to run even on CPU?
No, I tried it multiple times, but a lot of times it failed on this:
```
Error waiting for container: container 42ccbccd37a0619ea8f61cafe7389e192470da2d6e1f8e32f302d2e765a866b3: driver "overlay2" failed to remove root filesystem: unlinkat /var/lib/docker/overlay2/3e1e61e20a4e4676cb1c8e4bfdf1c9bd799b7bb775181ce553413047d74afe21: read-only file system
```
I googled it and found a tip to restart the docker, so I did that, but it didn't help. I haven't yet experienced such an issue with a Docker and a Mac. But maybe it's still possible to run it on Mac, I haven't really spent a lot of time trying to solve that.

**Comment by sokovninn** [comment]
Okay, let's make a note and add support for Mac Docker in the future if necessary.

**Pull request merged**